### ISSUE TITLE
[Bug] MSAL Login does not work after upgrading to winappsdk 1.6

### ISSUE THREAD

**Issue by cconner100** [AzureAD/microsoft-authentication-library-for-dotnet#4933]
### Library version used

<PackageVersion Include="Microsoft.Identity.Client" Version="4.63.0" /> <PackageVersion Include="Microsoft.Identity.Client.Extensions.Msal" Version="4.63.0" /> <PackageVersion Include="Microsoft.Identity.Client.Broker" Version="4.63.0" /> <PackageVersion Include="Microsoft.Identity.Client.Desktop" Version="4.63.0" /> <PackageVersion Include="Microsoft.Graph" Version="5.56.0" />

### .NET version

.net 8.300. Windows App SDK 1.6

### Scenario

PublicClient - desktop app

### Is this a new or an existing app?

None

### Issue description and reproduction steps

When our app was targeting WinAppSDK 1.5 the login would bring up the Browser login dialog box and allow us to login.  After upgrading to WinAppSDK 1.6 we get the following error message and no Browser will pop up

AuthoritySignIn https://ziidmsapp.b2clogin.com/tfp/ziidmsapp.onmicrosoft.com/B2C_1A_signup_signin
AuthorityRedirectUrl msal3e2fe1b7-910e-46d2-aa0b-b1da7755d458://auth
'ZiiDMSApp.exe' (CoreCLR: clrhost): Loaded 'E:\ZiiDMS-NextGen\Client\Desktop\ZiiDMSApp\ZiiDMSApp\bin\Debug\net8.0-windows10.0.22621\AppX\Uno.Core.Extensions.Logging.Singleton.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'ZiiDMSApp.exe' (CoreCLR: clrhost): Loaded 'E:\ZiiDMS-NextGen\Client\Desktop\ZiiDMSApp\ZiiDMSApp\bin\Debug\net8.0-windows10.0.22621\AppX\CommonServiceLocator.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
PermissionsScreenName is null or empty SetPermissionControls will not be called
'ZiiDMSApp.exe' (CoreCLR: clrhost): Loaded 'E:\ZiiDMS-NextGen\Client\Desktop\ZiiDMSApp\ZiiDMSApp\bin\Debug\net8.0-windows10.0.22621\AppX\Microsoft.Identity.Client.Extensions.Msal.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'ZiiDMSApp.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.8\System.IO.FileSystem.Watcher.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'ZiiDMSApp.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.8\System.Collections.Specialized.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'ZiiDMSApp.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App\8.0.8\System.Security.Cryptography.ProtectedData.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
MSAL: Always False MSAL 4.63.0.0 MSAL.NetCore .NET 8.0.8 Microsoft Windows 10.0.22631 [2024-09-18 05:48:20Z] [Internal cache] Clearing user token cache accessor. 
MSAL: Always False MSAL 4.63.0.0 MSAL.NetCore .NET 8.0.8 Microsoft Windows 10.0.22631 [2024-09-18 05:48:20Z] [Internal cache] Total number of cache partitions found while getting refresh tokens: 0 
MSAL: Info False MSAL 4.63.0.0 MSAL.NetCore .NET 8.0.8 Microsoft Windows 10.0.22631 [2024-09-18 05:48:20Z - c8b05dd3-8834-4c40-b8c6-4c2718affe93] Skipping Instance discovery for B2C authority because it is not supported. 
MSAL: Info False MSAL 4.63.0.0 MSAL.NetCore .NET 8.0.8 Microsoft Windows 10.0.22631 [2024-09-18 05:48:20Z - c8b05dd3-8834-4c40-b8c6-4c2718affe93] [Instance Discovery] Skipping Instance discovery for non-AAD authority.  
MSAL: Info False MSAL 4.63.0.0 MSAL.NetCore .NET 8.0.8 Microsoft Windows 10.0.22631 [2024-09-18 05:48:20Z] Found 0 cache accounts and 0 broker accounts 
MSAL: Info False MSAL 4.63.0.0 MSAL.NetCore .NET 8.0.8 Microsoft Windows 10.0.22631 [2024-09-18 05:48:20Z] Returning 0 accounts 
MSAL: Always False MSAL 4.63.0.0 MSAL.NetCore .NET 8.0.8 Microsoft Windows 10.0.22631 [2024-09-18 05:48:20Z] [Internal cache] Clearing user token cache accessor. 
MSAL: Always False MSAL 4.63.0.0 MSAL.NetCore .NET 8.0.8 Microsoft Windows 10.0.22631 [2024-09-18 05:48:20Z] [Internal cache] Total number of cache partitions found while getting refresh tokens: 0 
MSAL: Info False MSAL 4.63.0.0 MSAL.NetCore .NET 8.0.8 Microsoft Windows 10.0.22631 [2024-09-18 05:48:20Z - 7cba51d7-3401-482e-adcc-44d1dcd6f61d] Skipping Instance discovery for B2C authority because it is not supported. 
MSAL: Info False MSAL 4.63.0.0 MSAL.NetCore .NET 8.0.8 Microsoft Windows 10.0.22631 [2024-09-18 05:48:20Z - 7cba51d7-3401-482e-adcc-44d1dcd6f61d] [Instance Discovery] Skipping Instance discovery for non-AAD authority.  
MSAL: Info False MSAL 4.63.0.0 MSAL.NetCore .NET 8.0.8 Microsoft Windows 10.0.22631 [2024-09-18 05:48:20Z] Found 0 cache accounts and 0 broker accounts 
MSAL: Info False MSAL 4.63.0.0 MSAL.NetCore .NET 8.0.8 Microsoft Windows 10.0.22631 [2024-09-18 05:48:20Z] Returning 0 accounts 
MSAL: Info False MSAL 4.63.0.0 MSAL.NetCore .NET 8.0.8 Microsoft Windows 10.0.22631 [2024-09-18 05:48:20Z - a3957b6e-b482-44c7-8a4c-a7218dbcb4c2] MSAL MSAL.NetCore with assembly version '4.63.0.0'. CorrelationId(a3957b6e-b482-44c7-8a4c-a7218dbcb4c2) 
MSAL: Info False MSAL 4.63.0.0 MSAL.NetCore .NET 8.0.8 Microsoft Windows 10.0.22631 [2024-09-18 05:48:20Z - a3957b6e-b482-44c7-8a4c-a7218dbcb4c2] === AcquireTokenSilent Parameters === 
MSAL: Info False MSAL 4.63.0.0 MSAL.NetCore .NET 8.0.8 Microsoft Windows 10.0.22631 [2024-09-18 05:48:20Z - a3957b6e-b482-44c7-8a4c-a7218dbcb4c2] LoginHint provided: False 
MSAL: Info False MSAL 4.63.0.0 MSAL.NetCore .NET 8.0.8 Microsoft Windows 10.0.22631 [2024-09-18 05:48:20Z - a3957b6e-b482-44c7-8a4c-a7218dbcb4c2] Account provided: False 
MSAL: Info False MSAL 4.63.0.0 MSAL.NetCore .NET 8.0.8 Microsoft Windows 10.0.22631 [2024-09-18 05:48:20Z - a3957b6e-b482-44c7-8a4c-a7218dbcb4c2] ForceRefresh: False 
MSAL: Info False MSAL 4.63.0.0 MSAL.NetCore .NET 8.0.8 Microsoft Windows 10.0.22631 [2024-09-18 05:48:20Z - a3957b6e-b482-44c7-8a4c-a7218dbcb4c2] 
=== Request Data ===
Authority Provided? - True
Scopes - https://XXXX/access_as_user
Extra Query Params Keys (space separated) - 
ApiId - AcquireTokenSilent
IsConfidentialClient - False
SendX5C - False
LoginHint ? False
IsBrokerConfigured - False
HomeAccountId - False
CorrelationId - a3957b6e-b482-44c7-8a4c-a7218dbcb4c2
UserAssertion set: False
LongRunningOboCacheKey set: False
Region configured: 
 
MSAL: Info False MSAL 4.63.0.0 MSAL.NetCore .NET 8.0.8 Microsoft Windows 10.0.22631 [2024-09-18 05:48:20Z - a3957b6e-b482-44c7-8a4c-a7218dbcb4c2] === Token Acquisition (SilentRequest) started:
	 Scopes: https://XXXXX/access_as_user 
'ZiiDMSApp.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.8\System.Reflection.Metadata.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
MSAL: Error False MSAL 4.63.0.0 MSAL.NetCore .NET 8.0.8 Microsoft Windows 10.0.22631 [2024-09-18 05:48:20Z - a3957b6e-b482-44c7-8a4c-a7218dbcb4c2] Exception type: Microsoft.Identity.Client.MsalUiRequiredException
, ErrorCode: user_null
HTTP StatusCode 0
CorrelationId a3957b6e-b482-44c7-8a4c-a7218dbcb4c2
To see full exception details, enable PII Logging. See https://aka.ms/msal-net-logging
   at Microsoft.Identity.Client.Internal.Requests.Silent.SilentRequest.ExecuteAsync(CancellationToken cancellationToken)
   at Microsoft.Identity.Client.Internal.Requests.RequestBase.<>c__DisplayClass11_1.<<RunAsync>b__1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.Identity.Client.Utils.StopwatchService.MeasureCodeBlockAsync(Func`1 codeBlock)
   at Microsoft.Identity.Client.Internal.Requests.RequestBase.RunAsync(CancellationToken cancellationToken)
 
Exception thrown: 'Microsoft.Identity.Client.MsalUiRequiredException' in System.Private.CoreLib.dll
MSAL: Info False MSAL 4.63.0.0 MSAL.NetCore .NET 8.0.8 Microsoft Windows 10.0.22631 [2024-09-18 05:48:20Z - 110bfb27-5e09-4fd8-8b2a-7a74a85ec1b6] MSAL MSAL.NetCore with assembly version '4.63.0.0'. CorrelationId(110bfb27-5e09-4fd8-8b2a-7a74a85ec1b6) 
MSAL: Info False MSAL 4.63.0.0 MSAL.NetCore .NET 8.0.8 Microsoft Windows 10.0.22631 [2024-09-18 05:48:20Z - 110bfb27-5e09-4fd8-8b2a-7a74a85ec1b6] === InteractiveParameters Data ===
LoginHint provided: False
User provided: False
UseEmbeddedWebView: NotSpecified
ExtraScopesToConsent: 
Prompt: login
HasCustomWebUi: False 
MSAL: Info False MSAL 4.63.0.0 MSAL.NetCore .NET 8.0.8 Microsoft Windows 10.0.22631 [2024-09-18 05:48:20Z - 110bfb27-5e09-4fd8-8b2a-7a74a85ec1b6] 
=== Request Data ===
Authority Provided? - True
Scopes - https:/XXXXXXX/access_as_user
Extra Query Params Keys (space separated) - 
ApiId - AcquireTokenInteractive
IsConfidentialClient - False
SendX5C - False
LoginHint ? False
IsBrokerConfigured - False
HomeAccountId - False
CorrelationId - 110bfb27-5e09-4fd8-8b2a-7a74a85ec1b6
UserAssertion set: False
LongRunningOboCacheKey set: False
Region configured: 
 
MSAL: Info False MSAL 4.63.0.0 MSAL.NetCore .NET 8.0.8 Microsoft Windows 10.0.22631 [2024-09-18 05:48:20Z - 110bfb27-5e09-4fd8-8b2a-7a74a85ec1b6] === Token Acquisition (InteractiveRequest) started:
	 Scopes: https://ZiiDMSAPP.onmicrosoft.com/ZiiDMSWebApi/access_as_user 
MSAL: Info False MSAL 4.63.0.0 MSAL.NetCore .NET 8.0.8 Microsoft Windows 10.0.22631 [2024-09-18 05:48:20Z - 110bfb27-5e09-4fd8-8b2a-7a74a85ec1b6] [Instance Discovery] Skipping Instance discovery for non-AAD authority.  
MSAL: Error False MSAL 4.63.0.0 MSAL.NetCore .NET 8.0.8 Microsoft Windows 10.0.22631 [2024-09-18 05:48:20Z - 110bfb27-5e09-4fd8-8b2a-7a74a85ec1b6] Exception type: System.IO.FileNotFoundException
To see full exception details, enable PII Logging. See https://aka.ms/msal-net-logging
   at Microsoft.Identity.Client.Desktop.WebView2WebUi.WebView2WebUiFactory.IsWebView2Available()
   at Microsoft.Identity.Client.Desktop.WebView2WebUi.WebView2WebUiFactory.CreateAuthenticationDialog(CoreUIParent coreUIParent, WebViewPreference useEmbeddedWebView, RequestContext requestContext)
   at Microsoft.Identity.Client.Internal.AuthCodeRequestComponent.CreateWebAuthenticationDialog()
   at Microsoft.Identity.Client.Internal.AuthCodeRequestComponent.FetchAuthCodeAndPkceVerifierAsync(CancellationToken cancellationToken)
   at Microsoft.Identity.Client.Internal.Requests.InteractiveRequest.GetTokenResponseAsync(CancellationToken cancellationToken)
   at Microsoft.Identity.Client.Internal.Requests.InteractiveRequest.ExecuteAsync(CancellationToken cancellationToken)
   at Microsoft.Identity.Client.Internal.Requests.RequestBase.<>c__DisplayClass11_1.<<RunAsync>b__1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.Identity.Client.Utils.StopwatchService.MeasureCodeBlockAsync(Func`1 codeBlock)
   at Microsoft.Identity.Client.Internal.Requests.RequestBase.RunAsync(CancellationToken cancellationToken)
 
Exception thrown: 'System.IO.FileNotFoundException' in System.Private.CoreLib.dll
Could not load file or assembly 'Microsoft.Web.WebView2.Core, Version=1.0.864.35, Culture=neutral, PublicKeyToken=2a8ab48044d2601e'. The system cannot find the file specified.

### Expected behavior

Expect it to work like the WinAppSDK 1.5 version which popups the login window

### Identity provider

Azure B2C Custom Policy

**Comment by cconner100** [comment]
If i change my code to use WithDefaultREdirectUri and not with the browserredirect it works.  But it would be nice to have the older code work again

**Comment by CameronVetter** [comment]
Any plan for this?  This is a complete showstopper for existing applications.  I can't believe this isn't urgent enough that your team hasn't even bothered to respond to this issue.

**Comment by localden** [comment]
@CameronVetter investigation is ongoing.

**Comment by panta2705** [comment]
Any news here?
We have same issue that msal library requires old version of Microsoft.Web.WebView2 on dotnet 9. Because latest (v9.0.10) maui library Microsoft.Maui.Core now includes new Microsoft.WindowsAppSDK as well as Microsoft.Web.WebView2 directly now so our sign in process is broken and our migration to dotnet 9 blocked by this
![image](https://github.com/user-attachments/assets/4f23f580-e62d-4d79-a1ef-fdadcb70a9b3)

**Comment by localden** [comment]
@panta2705 - can you walk me through your repro steps, along with the snippet of code that you are using to authenticate?

I am testing this as we speak with MAUI and .NET 9 and so far not running into the issue you are calling out.

**Comment by localden** [comment]
@cconner100 @CameronVetter can you also please provide the code snippets that you are using that are triggering this? From my testing it seems contained to B2C scenarios.

**Comment by CameronVetter** [comment]
I can confirm my scenario is a B2C application.  Here are my packages:

```
<ItemGroup>
        <PackageReference Include="Azure.Storage.Queues" Version="12.20.1" />
        <PackageReference Include="BlazorTransitionableRoute" Version="4.0.0" />
        <PackageReference Include="CommunityToolkit.Maui" Version="9.1.1" />
        <PackageReference Include="CommunityToolkit.Mvvm" Version="8.3.2" />
        <PackageReference Include="Kebechet.Maui.RevenueCat.InAppBilling" Version="4.5.3" />
        <PackageReference Include="Microsoft.ApplicationInsights.WorkerService" Version="2.22.0" />
        <PackageReference Include="Microsoft.AspNetCore.Components" Version="8.0.10" />
        <PackageReference Include="Microsoft.AspNetCore.Components.Authorization" Version="8.0.10" />
        <PackageReference Include="Microsoft.AspNetCore.Components.WebView.Maui" Version="8.0.93" />
        <PackageReference Include="Microsoft.Extensions.Logging.ApplicationInsights" Version="2.22.0" />
        <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="8.0.1" />
        <PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="8.0.1" />
        <PackageReference Include="Microsoft.Extensions.Logging.abstractions" Version="8.0.2" />
        <PackageReference Include="Microsoft.Identity.Client" Version="4.66.1" />
        <PackageReference Include="Microsoft.Identity.Client.Desktop" Version="4.66.1" Condition="'$(TargetFramework)' == 'net8.0-windows10.0.19041.0'" />
        <PackageReference Include="Microsoft.Identity.Client.Extensions.Msal" Version="4.66.1" />
        <PackageReference Include="Microsoft.Maui.Controls" Version="8.0.93" />
        <PackageReference Include="Microsoft.Maui.Controls.Compatibility" Version="8.0.93" />
        <PackageReference Include="Microsoft.Web.WebView2" Version="1.0.2849.39" Condition="'$(TargetFramework)' == 'net8.0-windows10.0.19041.0'" />
        <PackageReference Include="Microsoft.WindowsAppSDK" Version="1.6.240923002" Condition="'$(TargetFramework)' == 'net8.0-windows10.0.19041.0'" />
        <PackageReference Include="MudBlazor" Version="7.15.0" />
        <PackageReference Include="MudBlazor.Markdown" Version="7.14.0" />
        <PackageReference Include="Newtonsoft.Json" Version="13.0.3" />
        <PackageReference Include="Polly" Version="8.4.2" />
        <PackageReference Include="RestSharp" Version="112.1.0" />
        <PackageReference Include="RestSharp.Serializers.NewtonsoftJson" Version="112.1.0" />
        <PackageReference Include="System.Diagnostics.PerformanceCounter" Version="8.0.1" />
    </ItemGroup>
```

and the code that triggers the error is (greatly simplified):

```
            PublicClientApplication = PublicClientApplicationBuilder
                .WithWindowsEmbeddedBrowserSupport()
                .WithRedirectUri($"https://bgcopilot.b2clogin.com/oauth2/nativeclient")
                .Build();

                return await PublicClientApplication.AcquireTokenInteractive(AzureADB2CConfig.Scopes)
                   .WithUseEmbeddedWebView(true)
                   .ExecuteAsync();
```

**Comment by localden** [comment]
I am running this as a test:

```
var app = PublicClientApplicationBuilder.Create("foobar")
   .WithWindowsEmbeddedBrowserSupport()
   .WithRedirectUri($"https://bgcopilot.b2clogin.com/oauth2/nativeclient")
   .Build();

var t = await app.AcquireTokenInteractive(new string[]{ "user.read" })
   .WithUseEmbeddedWebView(true)
   .ExecuteAsync();
```

I do not see any errors and the web view correctly shows. This targets `net-8.0`, by the way - `net9.0` has a different blocking bug (https://github.com/dotnet/maui/issues/25883) that prevents me from testing it.

What version of Windows are you running? Can you also please upgrade to the latest MSAL?

**Comment by CameronVetter** [comment]
I'm running the latest version of Windows 11:

![image](https://github.com/user-attachments/assets/0f6c575c-d3a2-47d1-bd62-c48121354abb)

I tried updating to 4.66.2 which is latest and still have the same filenotfound exception

**Comment by panta2705** [comment]
@localden Here is some simplified code snippets, I can try making a sample project with the issue if needed

```
public AuthenticationProvider()
{
	var builder = PublicClientApplicationBuilder
		.Create(_config.ClientId)
		.WithIosKeychainSecurityGroup(_config.IOSKeychainSecurityGroup)
		.WithB2CAuthority(_config.AuthoritySignInUp)
		.WithRedirectUri($"msal{_config.ClientId}://auth");
		
	#if WINDOWS
		builder.WithWindowsEmbeddedBrowserSupport();
	#endif

		_pca = builder.Build();

	#if WINDOWS
		var storageProperties =
			new StorageCreationPropertiesBuilder("msal-cache", storageSettings.MsalCacheDirectory)
				.Build();

		// This hooks up the cross-platform cache into MSAL
		var cacheHelper = MsalCacheHelper.CreateAsync(storageProperties);
		cacheHelper.Result.RegisterCache(_pca.UserTokenCache);
	#endif
}

public Task<AuthenticationResult> SignInAsync(string emailHint)
{
    return _pca!.AcquireTokenInteractive(_config!.Scopes)
		.WithB2CAuthority(_config.AuthoritySignInUp)
#if ANDROID || IOS
        .WithUseEmbeddedWebView(true)
#endif
        .WithPrompt(Prompt.ForceLogin)
        .WithLoginHint(emailHint)
        .WithExtraQueryParameters(_extraQueryParams)
        .ExecuteAsync()
        .WrapMsalExceptionsAsync();
}
```

Respective libs used that produce error
```
<PackageVersion Include="CommunityToolkit.Maui" Version="9.1.1" />
<PackageVersion Include="Microsoft.Identity.Client" Version="4.66.2" />
<PackageVersion Include="Microsoft.Identity.Client.Desktop" Version="4.66.2" />
<PackageVersion Include="Microsoft.Identity.Client.Extensions.Msal" Version="4.66.2" />
<PackageVersion Include="Microsoft.Maui.Controls" Version="9.0.10" />
<PackageVersion Include="Microsoft.Maui.Controls.Compatibility" Version="9.0.10" />
<PackageVersion Include="Microsoft.AspNetCore.Components.WebView.Maui" Version="9.0.10" />
```

Respective libs used that are working
```
<PackageVersion Include="CommunityToolkit.Maui" Version="9.1.1" />
<PackageVersion Include="Microsoft.Identity.Client" Version="4.66.2" />
<PackageVersion Include="Microsoft.Identity.Client.Desktop" Version="4.66.2" />
<PackageVersion Include="Microsoft.Identity.Client.Extensions.Msal" Version="4.66.2" />
<PackageVersion Include="Microsoft.Maui.Controls" Version="8.0.100" />
<PackageVersion Include="Microsoft.Maui.Controls.Compatibility" Version="8.0.100" />
<PackageVersion Include="Microsoft.AspNetCore.Components.WebView.Maui" Version="8.0.100" />
```

Windows version
![image](https://github.com/user-attachments/assets/a71e99cb-0995-4b9a-b7bc-8930f9d42d66)

**Comment by localden** [comment]
Thank you, folks - will be running a few more tests today to see if I can reproduce this reliably.

**Comment by ancker1** [comment]
We observe the same problem in our applications

**Comment by CameronVetter** [comment]
Any updates @localden?  I'm still seeing this problem and blocked from releasing on Windows.  Were you able to repro this locally?  It's been many months without any progress on this critical showstopper!  What is the timeline for getting this fixed?

**Comment by localden** [comment]
We're still working on a repro for this. @CameronVetter @mibinoob @ancker1 - what would be most helpful at this time is if you have a "minimum reproducible project." It can be a simple app that shows the behavior reliably, that you can attach to the issue.

**Comment by unrateddi** [comment]
Hi @localden , you can check my issue #5015 . The text before repro steps and the repro steps mention how to reproduce it easily in the client samples of this repo.

**Comment by unrateddi** [comment]
Hi @localden you can use my sample to reproduce the issue with the old WebView version referenced by ms-auth. It is just a fork of ms-identity-dotnetcore-maui samples, upgraded to net9 and updated all nuget packages to latest. Also added the csproj workaround to be able to build in windows because of this [issue](https://github.com/dotnet/windowsdesktop/issues/4789) (described in more detail in my issue #5015 ). 
[repro-version-notfound](https://github.com/unrateddi/repro-ms-identity-dotnetcore-maui/tree/repro-version-notfound)

Please use the MauiAppBasic project, fill-in the appsettings.json with auth info. The app should be run on Windows platform as the issue is only relevant for Windows. Make sure that you select the **Use Embedded** checkbox in the app before you hit "Sign In".
![image](https://github.com/user-attachments/assets/e4ed97a7-4fa3-47a4-8ad8-10a0eb33c640)

You should end-up with the error:
![image](https://github.com/user-attachments/assets/a26ce8e0-5a5a-4035-bd4f-67cfb36a2b14)

To add, you can also look into this [branch](https://github.com/unrateddi/repro-ms-identity-dotnetcore-maui/tree/repro-authworks-mauiwebviewfails), which directly references the version the error mentions and the auth works now, but we are unable to use any WebView control in MAUI-Windows because it looks up for a newer version giving the following error. After successfully login with "Use Embedded" option the app goes to the UserView page which I have place a WebView control.

![image](https://github.com/user-attachments/assets/82d92e3b-5d38-407e-9e0a-7ce20ab92583)

**Comment by CameronVetter** [comment]
Has anyone come up with a workaround for this?  As of yesterday I now have two apps that are blocked due to this bug...

**Comment by localden** [comment]
@iulico-1 @ashok672 seems like the issue is indeed caused by what @unrateddi and @espenrl called out (older version of `Microsoft.Web.WebView2` dependency). We should prioritize updating it to avoid the current issue in WinUI/.NET 9 applications.

**Comment by lukaszzielnica** [comment]
When a bugfix is planned to be released?

**Comment by localden** [comment]
@lukaszzielnica - will delegate this decision to @iulico-1 and @ashok672, as I am no longer in the security side of the house.

**Comment by espenrl** [comment]
@lukaszzielnica @iulico-1 @ashok672 Can this issue be prioritized as this hinder the use of MAUI on Windows with .NET 9. All three technologies are Microsoft products within active support. It has been known since September 18th and a fix would be much welcomed.

**Comment by ancker1** [comment]
> [@lukaszzielnica](https://github.com/lukaszzielnica) [@iulico-1](https://github.com/iulico-1) [@ashok672](https://github.com/ashok672) Can this issue be prioritized as this hinder the use of MAUI on Windows with .NET 9. All three technologies are Microsoft products within active support. It has been known since September 18th and a fix would be much welcomed.

+1
I hope you are able to prioritize this issue.

**Comment by CameronVetter** [comment]
+2
Please prioritize, and I certainly hope you will fix the testing gap that allowed this to ship in a broken state, it clearly was never tested.

**Comment by dtentori** [comment]
same error here with MAUI Hybrid Blazor + .NET 9 + Windows target. Please prioriize

**Comment by SimonDutton** [comment]
I have raised an issue on MS Developer Community as I am also getting this problem since upgrading to .Net 9 and it includes an Example .Net 8 app that works and a .Net 9 one that does not.

https://developercommunity.visualstudio.com/t/DotNet-9-PublicClientApplication-Microso/10841011

**Comment by ancker1** [comment]
@localden @iulico-1 @ashok672 are you able to provide an update on this issue?

**Comment by DharshanBJ** [comment]
@unrateddi , we haven't been able to reproduce this issue, even with the sample repo you provided (.NET 9/WinAppSDK 1.6). Are you still encountering the problem?

As mentioned here - https://www.nuget.org/packages/Microsoft.Identity.Client.Desktop#dependencies-body-tab, the required version of WebView2 is >= 1.0.864.35, so it should work for the newer versions of webview2 as well, we're unsure what might be causing this issue.

Since 1.0.864.35 is an very older version of WebView2, we're working on upgrading it to a newer version, but we're not sure yet if that will resolve the issue.

**Comment by unrateddi** [comment]
@DharshanBJ , yes this is still reproducible in my sample repo even with the latest packages. I just pushed a commit with all nugets updated.

Please confirm you are trying to reproduce it on the correct branch as mentioned in my post. The branch name is [repro-version-notfound](https://github.com/unrateddi/repro-ms-identity-dotnetcore-maui/tree/repro-version-notfound). Repro should be pretty easy on MauiAppBasic project like fill in appsettings.json with auth info, build, run, click "Use Embedded" checkbox, click "Sign in" button and then the exception should be thrown mentioning that version 1.0.864.35 of the dll is not found.

Also you might need to enable System.IO.FileNotFoundException in Exception Settings in Visual Studio to get proper exception insights.

**Comment by SimonDutton** [comment]
I have added a simple VS Solution containing a .Net8 and a .Net9 version of the same project.  The .Net8 works but the .Net9 does not.
[MauiMSAL-Test.zip](https://github.com/user-attachments/files/18798538/MauiMSAL-Test.zip)

if you try the .Net9 version you get this error : 'Could not load file or assembly 'Microsoft.Web.WebView2.Core, Version=1.0.864.35, Culture=neutral, PublicKeyToken=2a8ab48044d2601e'. The system cannot find the file specified.'

![Image](https://github.com/user-attachments/assets/383a4d5e-1e48-42e6-be15-96a6018f51ab)

**Comment by Forestbrook** [comment]
We also still have this issue, please fix it ASAP!
This is an urgent issue for us, because we need nuget packages with fixes only available in the .NET9 version in our app.

We get this exception message when logging on:
`Could not load file or assembly 'Microsoft.Web.WebView2.Core, Version=1.0.864.35, Culture=neutral, PublicKeyToken=2a8ab48044d2601e'. The system cannot find the file specified.`

This is interestingly, because the Microsoft.Client.IdentityDesktop package states it needs Microsoft.Web.WebView2 >= 1.0.864.35, so it should work with Microsoft.Web.WebView2 version 1.0.2792.45

The problem in both situations is the DLL reference:
1.0.864.35 is using **Microsoft.Web.WebView2.Core**
1.0.2792.45 and later is using **Microsoft.Web.WebView2.Core.Projection**

So please fix the direct reference to the dll!

**Comment by dtentori** [comment]
I confirm reported issues and comments. We're developing a project in MAUI Hybrid Blazor with "net9" with MSAL interactive auth and this bug forces us to target "net8" in the whole project.
Please solve, thanks

**Comment by unrateddi** [comment]
Hello @DharshanBJ , any updates on this? We are 3 months in .NET9 with having to decide if we want working ms-auth or webview in MAUI Windows.

**Comment by ancker1** [comment]
@localden @iulico-1 @ashok672 any news? 

For us developers it is hard to understand how this issue is not prioritized higher.

**Comment by DharshanBJ** [comment]
> I have added a simple VS Solution containing a .Net8 and a .Net9 version of the same project. The .Net8 works but the .Net9 does not. [MauiMSAL-Test.zip](https://github.com/user-attachments/files/18798538/MauiMSAL-Test.zip)
> 
> if you try the .Net9 version you get this error : 'Could not load file or assembly 'Microsoft.Web.WebView2.Core, Version=1.0.864.35, Culture=neutral, PublicKeyToken=2a8ab48044d2601e'. The system cannot find the file specified.'
> 
> ![Image](https://github.com/user-attachments/assets/383a4d5e-1e48-42e6-be15-96a6018f51ab)

Thanks simon for sharing your sample, we are able to repro this issue using your sample app now.
I'm actively working on implementing a fix for this.

**Comment by DharshanBJ** [comment]
This seems to be a bug with the Microsoft.Web.Webview2 package as reported here - https://github.com/MicrosoftEdge/WebView2Feedback/issues/4784

We are still testing this with older .net versions along with different winapp versions to ensure that the Microsoft.Web.Webview2 package that we will be upgrading to won't regress any other scenario

**Comment by DharshanBJ** [comment]
This indeed seems to be a bug with webview2 on .Net Maui 9. We have opened a bug on Webview2 team - https://github.com/MicrosoftEdge/WebView2Feedback/issues/5142

**Comment by mibinoob** [comment]
For those who have not noticed:
Upgrading SDK and Webview to newest versions and adding `<WebView2EnableCsWinRTProjection>false</WebView2EnableCsWinRTProjection>` to csproj files where you are using the packages seem to be a working quick fix. No promises, but it worked for me NET8, MSAL.Desktop on WinUI3. Hope it works for you as well!

**Comment by unrateddi** [comment]
It doesn't work with .NET9 only with .NET8. In .NET9 it fixes the ms-auth webview but maui webview controls in pages throw this error.

![Image](https://github.com/user-attachments/assets/270a6cfb-1c9f-4d91-ab7e-414cfdd8b0e7)

**Comment by espenrl** [comment]
`Microsoft.Maui.dll` has a hard reference on `Microsoft.Web.WebView2.Core.Projection.dll`

<img width="358" alt="Image" src="https://github.com/user-attachments/assets/e38889b0-604c-432d-9c64-b325b3f878b6" />

**Comment by DharshanBJ** [comment]
This issue is related to a bug in WebView2, as detailed in this [GitHub issue](https://github.com/MicrosoftEdge/WebView2Feedback/issues/5142). Until WebView2 releases an updated package with a fix, the MSAL.NET team is unable to take any action.

**Comment by VaibhavMethuku01** [comment]
I had created a POC MAUI .Net 9 application with adding a webview in that, Even in that I'm not able to open the webview in the windows .?

Do we have any work around for that.?

@DharshanBJ

**Comment by mattleibow** [comment]
Just added some more info here: https://github.com/dotnet/maui/issues/28717#issuecomment-2872993670

> This all stems from the main issue. the net9.0-windows TFM is used for all the windows apps, winforms, wpf, winui and maui. We can reduce this into 2 separate worlds: those that use WinRT (winui/maui) and those that don't (winforms/wpf).
> 
> This may seem trivial, but the main issue is that people using WinRT are using the Microsoft.Web.WebView2.Core.Projection.dll that then uses COM to talk to the native Microsoft.Web.WebView2.Core.dll. The rest are using WINMD which uses the Microsoft.Web.WebView2.Core.winmd file to talk WINMD to the native Microsoft.Web.WebView2.Core.dll.
> 
> Now, this main issue arises when using the Microsoft.Identity.Client.Desktop nuget. This nuget was built using the NON-WINRT nuget, so this means that when it is looking for Microsoft.Web.WebView2.Core, it is really looking for Microsoft.Web.WebView2.Core.winmd. When running in a .NET MAUI app, that dll DOES exist, but it is a native binary and we cannot use it directly and MUST use it via the Microsoft.Web.WebView2.Core.Projection.dll.
> 
> So, what does this mean? Well, the MSAL library will have to do something that the WebView2 people did and have 2 lib folders, one for the WINMD way and one for the WINRT way.

**Comment by DharshanBJ** [comment]
We are working on adding Modern TFM support for msal.net. This is aimed at fixing the winrt issue that MAUI 9 is hitting.
The ETA for releasing this fix is 2nd week of July.

**Comment by mibinoob** [comment]
@DharshanBJ and what about the other frameworks which is also suffering from this issue? 

To elaborate: 
Running e.g. a WinUI3 application with .NET8 is also affected by this issue, will this also be covered in the July patch or is it solely MAUI?

**Comment by DharshanBJ** [comment]
> [@DharshanBJ](https://github.com/DharshanBJ) and what about the other frameworks which is also suffering from this issue?
> 
> To elaborate: Running e.g. a WinUI3 application with .NET8 is also affected by this issue, will this also be covered in the July patch or is it solely MAUI?

We will be covering other frameworks as well in the patch.

**Comment by DharshanBJ** [comment]
Update: We are actively working on this feature work, the eta will slip by a couple of weeks, as we are running into some issues integrating with winui3, more details here - https://github.com/microsoft/microsoft-ui-xaml/issues/10647

We plan to have a release out with the winui3 support by the end of July.

**Comment by DharshanBJ** [comment]
The work for adding winui3 support for msal.net desktop package is now in PR - https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/5411/files

**Comment by DharshanBJ** [comment]
[msal_packages.zip](https://github.com/user-attachments/files/21740620/msal_packages.zip)

The attached ZIP file contains preview packages with WinUI 3 support. For WinApp SDK packaged applications or MAUI 9 applications, we recommend using the Microsoft.Identity.Client.Desktop.WinUI3.nupkg package instead of Microsoft.Identity.Client.Desktop.nupkg.

While we continue working on getting the corresponding PR merged and an official release published, we encourage customers to try out these preview packages and test their authentication flows.

**Comment by DevDachi** [comment]
@DharshanBJ Thank you very much for your efforts to push this and providing these test packages.

I tested these packages for our .NET Maui 9.0 Windows App and it works perfectly (B2C auth). Waiting eagerly for the public version of these packages.

**Comment by espenrl** [comment]
It works. I've tested with MAUI v9 SR 10 using Windows on Arm.

**Comment by mibinoob** [comment]
Tested it for net8.0-windows10.0.19041.0 (Winx86) on a WinUI3 desktop application.
 
When trying to get the window prompting the user to log in using:
 
_publicClientApplication.AcquireTokenInteractive(_scopes);
 
I get "Object reference not set to an instance of an object", with the stacktrace
 
at Microsoft.Identity.Client.Internal.AuthCodeRequestComponent.VerifyAuthorizationResult(AuthorizationResult authorizationResult, String originalState)
   at Microsoft.Identity.Client.Internal.AuthCodeRequestComponent.<FetchAuthCodeAndPkceInternalAsync>d__7.MoveNext()
   at Microsoft.Identity.Client.Internal.AuthCodeRequestComponent.<FetchAuthCodeAndPkceVerifierAsync>d__4.MoveNext()
   at Microsoft.Identity.Client.Internal.Requests.InteractiveRequest.<GetTokenResponseAsync>d__11.MoveNext()
   at Microsoft.Identity.Client.Internal.Requests.InteractiveRequest.<ExecuteAsync>d__9.MoveNext()
   at Microsoft.Identity.Client.Internal.Requests.RequestBase.<>c__DisplayClass11_1.<<RunAsync>b__1>d.MoveNext()
   at Microsoft.Identity.Client.Utils.StopwatchService.<MeasureCodeBlockAsync>d__4.MoveNext()
   at Microsoft.Identity.Client.Internal.Requests.RequestBase.<RunAsync>d__11.MoveNext()
   at Microsoft.Identity.Client.ApiConfig.Executors.PublicClientExecutor.<ExecuteAsync>d__2.MoveNext()
 
I tried to force it to use the prompt by adding .WithPrompt(Prompt.SelectAccount), but to no success.
 
I checked everything in debugging, but found no object not yet instantiated (note this worked previously).
 
It therefore seems the issue is not solved for our build as of yet.
 
Please ask if you need more information

**Comment by panta2705** [comment]
> Tested it for net8.0-windows10.0.19041.0 (Winx86) on a WinUI3 desktop application.
> 
> When trying to get the window prompting the user to log in using:
> 
> _publicClientApplication.AcquireTokenInteractive(_scopes);
> 
> I get "Object reference not set to an instance of an object", with the stacktrace
> 
> at Microsoft.Identity.Client.Internal.AuthCodeRequestComponent.VerifyAuthorizationResult(AuthorizationResult authorizationResult, String originalState) at Microsoft.Identity.Client.Internal.AuthCodeRequestComponent.d__7.MoveNext() at Microsoft.Identity.Client.Internal.AuthCodeRequestComponent.d__4.MoveNext() at Microsoft.Identity.Client.Internal.Requests.InteractiveRequest.d__11.MoveNext() at Microsoft.Identity.Client.Internal.Requests.InteractiveRequest.d__9.MoveNext() at Microsoft.Identity.Client.Internal.Requests.RequestBase.<>c__DisplayClass11_1.<b__1>d.MoveNext() at Microsoft.Identity.Client.Utils.StopwatchService.d__4.MoveNext() at Microsoft.Identity.Client.Internal.Requests.RequestBase.d__11.MoveNext() at Microsoft.Identity.Client.ApiConfig.Executors.PublicClientExecutor.d__2.MoveNext()
> 
> I tried to force it to use the prompt by adding .WithPrompt(Prompt.SelectAccount), but to no success.
> 
> I checked everything in debugging, but found no object not yet instantiated (note this worked previously).
> 
> It therefore seems the issue is not solved for our build as of yet.
> 
> Please ask if you need more information

We had similar issue with the same exception (but on dotnet 9) and noticed that it needs to run from main (UI) thread in order to work. You can try it out

**Comment by mibinoob** [comment]
@panta2705 Thanks for the answer. However, I have not found a way to specify what thread the method should execute from. The nearest I could find was a ".ConfigureAwait(true)", but this did little to solve the issue. 

How did you specify what thread to use when executing?

**Comment by panta2705** [comment]
@mibinoob 
In our maui app we do it similar to this
<img width="414" height="81" alt="Image" src="https://github.com/user-attachments/assets/27030477-0455-4178-9792-9042123f4089" />

**Comment by mibinoob** [comment]
@panta2705 I acutally just got it working, thanks though.

I discovered we used to start a separate Task to run it on, but as you said only main thread seem to do the trick.

I got it working now also.

Edit: To others with similar issue and no direct way of separating main threads from other threads. 
If you do a Task.Run(await work()), it does not run on the main thread, and the window will fail to open. 
Deleting the Task.Run, so you only have await work() now allows MSAL to open the window.

**Comment by espenrl** [comment]
> It works. I've tested with MAUI v9 SR 10 using Windows on Arm.

Quick note. I had to change parameter into `WithParentActivityOrWindow` from a window handle to the window object.

<img width="399" height="53" alt="Image" src="https://github.com/user-attachments/assets/141a2bed-db0c-4d20-b684-7c0bf9867d29" />

<img width="379" height="83" alt="Image" src="https://github.com/user-attachments/assets/7938357b-8089-4368-854f-de156eaf3d94" />

**Comment by DevDachi** [comment]
Thanks @DharshanBJ, PR is merged now. When can we expect this in the public nugets?

**Comment by stuccitto** [comment]
Using the winui3 support for msal.net desktop package that @DharshanBJ posted, I was able to get a net-9 maui example "working". I wanted the embedded browser to appear in the ContentPage on a MainPage but, instead of AcquireTokenInteractive  embedding the browser in the window passed to it via WithParentActivityOrWindow, its creating a new window and embedding the browser in that.  After some poking around with copilot, I got the following:

> Important: There is currently no supported way in MSAL.NET to force the embedded web view to use the existing app window in WinUI3/.NET MAUI. The embedded web view will always create its own window. The best you can do is set the parent window handle using .WithParentActivityOrWindow, which ensures the new window is properly parented, but it will not prevent the creation of a new window.
> If you need the web view to appear inside your main window, you must use a system browser or implement a custom authentication UI using a WebView control in your app.
> 

So is this really the expected behavior i.e. it creates a new window or am I not doing something correctly? Any insight/clarification would be appreciated.
Thanks

here's the code I'm using:

```
            PlatformConfig.Instance.ParentWindow = () =>
            {
                var mauiWindow = Microsoft.Maui.Controls.Application.Current?.Windows?.FirstOrDefault();
                var nativeWindow = mauiWindow?.Handler?.PlatformView as Microsoft.UI.Xaml.Window;
                return nativeWindow is null ? IntPtr.Zero : WindowNative.GetWindowHandle(nativeWindow);
            };

                PublicClientApplication = PublicClientApplicationBuilder.Create(ClientId)
                        .WithExperimentalFeatures() // this is for upcoming logger
                        .WithAuthority(LoginAuthority)
                        .WithLogging((level, message, pii) => Console.WriteLine($"{level}: {message}"), LogLevel.Verbose, true)
                        .WithRedirectUri($"msal{ClientId}://auth")
 #if WINDOWS
                    .WithWindowsEmbeddedBrowserSupport()
#endif
                .Build();
            }

            var AuthResult = await PublicClientApplication.AcquireTokenInteractive(appScopes)
                    .WithUseEmbeddedWebView(true)
                    .WithParentActivityOrWindow(PlatformConfig.Instance.ParentWindow)
                    .ExecuteAsync();

```

**Comment by vaibhavMethuku** [comment]
Hi @DharshanBJ, Can I have the sample integration for MSAL with the latest private nugets

**Comment by DharshanBJ** [comment]
we have now officially released the new package to nuget - https://www.nuget.org/packages/Microsoft.Identity.Client.Desktop.WinUI3/

**Comment by vaibhavMethuku** [comment]
I had updated to latest packages

`
				mPublicClientApplication = PublicClientApplicationBuilder.Create(options.ClientId)
																.WithAuthority(options.Authority)
																.WithRedirectUri(SpecialRedirectUri ?? $"msal{options.ClientId}://auth")
#if WINDOWS
																.WithWindowsEmbeddedBrowserSupport()
#endif
																.Build();
`

```
result = await mPublicClientApplication.AcquireTokenInteractive(mIndustrialAppsADOptions.Scopes)
					  .WithUseEmbeddedWebView(true)
					  .ExecuteAsync();

```

I'm facing the issue with the interactive login. 

{"Could not load file or assembly 'Microsoft.Web.WebView2.Core.Projection, Version=1.0.2903.40, Culture=neutral, PublicKeyToken=2a8ab48044d2601e'. The system cannot find the file specified.":"Microsoft.Web.WebView2.Core.Projection, Version=1.0.2903.40, Culture=neutral, PublicKeyToken=2a8ab48044d2601e"}

Please lemme know if need to any changes in this

FYI I'm working with the .Net 9

using this package

#if WINDOWS

using Microsoft.Identity.Client.Desktop.WinUI3;
#endif

@DharshanBJ @DevDachi

**Comment by mibinoob** [comment]
@vaibhavMethuku I think I encountered this issue earlier, and as I wrote:
It appears the window has to run on the main thread. This means if this is in some way (in the method or where the method is called) is not running on the main thread, for example, using Task.Run(LoginMethod()) will create a seperate thread and bring said error.

My initial thought is this is the case you have encountered. 
EDIT:
No, it actually does not work for me now. I had other tasks at hand and first got to updating now. It did work in the pre-build version, but I cannot get it to run now. @DharshanBJ Something has definitely gone wrong when using .NET Desktop.

EDIT2:
Did some debugging to ease solving the issue. If I look at the error under "TargetSite" I get

<img width="958" height="99" alt="Image" src="https://github.com/user-attachments/assets/4831cf14-685c-4451-b98f-cd8ca9b3249c" />

Additionally the IPublicClientApplication.AppConfig has a WebUiFactoryCreator<WithWindowsEmbeddedBrowserSupport>() which should return an IWebUIFactory but the reflected type also throws ReflectedType = '((System.Reflection.RuntimeMethodInfo)((Microsoft.Identity.Client.ApplicationConfiguration)((Microsoft.Identity.Client.ClientApplicationBase)_publicClientApplication).AppConfig).WebUiFactoryCreator.Method).ReflectedType' threw an exception of type 'System....

To me it seems like the Factory for using WebUI cannot be created.

The only thing I can remember having changed since I tested the preversion is upgrading to Windows11

I even tried downgrading from newest Webview version to 1.0.2903.40, but it did not help.

**Comment by vaibhavMethuku** [comment]
I tried running it on the main thread, but unfortunately had no success.

 On Android, I’m required to update to .NET 9 to support the new 16KB page size. Currently, I’m maintaining separate branches for .NET 8 and .NET 9 for Android and iOS, but managing two branches is becoming very difficult.

@DharshanBJ  @mibinoob

**Comment by mibinoob** [comment]
I got something working.
If you are running on Desktop you no longer use Microsoft.Identity.Client.Desktop you need to use Microsoft.Identity.Client.Desktop.WinUI3. 
Removing the reference and adding the other or in other ways ensuring you call the PublicApplicationBuilder with Microsoft.Identity.Client.Desktop.WinUI3 should make it work at least if you use WinUI3.
@vaibhavMethuku

**Comment by DharshanBJ** [comment]
@vaibhavMethuku Please refer to this sample project that SimonDutton had shared in this thread earlier, it contains 2 projects, one with net8 and one with net9, i have updated the net9 proj to use the WinUI3 package and it works as expected now

[MauiMSAL-Test.zip](https://github.com/user-attachments/files/23507296/MauiMSAL-Test.zip)

**Comment by Arslan007** [comment]
@DharshanBJ Its not working with latest LTS .NET10. Using new WinUI3 package but getting this error on build time.

```
Could not copy the file "Path\.nuget\packages\microsoft.maui.controls.compatibility\10.0.10\lib\net10.0-windows10.0.20348\Microsoft.Maui\Handlers\HybridWebView\HybridWebView.js" because it was not found.
Could not copy the file "Path\.nuget\packages\microsoft.maui.controls.core\10.0.10\lib\net10.0-windows10.0.20348\Microsoft.Maui\Handlers\HybridWebView\HybridWebView.js" because it was not found.
Could not copy the file "Path\.nuget\packages\microsoft.maui.core\10.0.10\lib\net10.0-windows10.0.20348\Microsoft.Maui\Handlers\HybridWebView\HybridWebView.js" because it was not found.
```

**Comment by DharshanBJ** [comment]
@Arslan007, Thanks for bringing this up! However, the issue you're encountering seems to be different from the one originally described in this ticket. To help us better track and address each issue individually, could you please create a new issue for your specific problem?
